### PR TITLE
Rename reboot order from `priority` to be `startupOrder`

### PR DESCRIPTION
**Pull request by yophilav**
Rename reboot order from `priority` to be `startupOrder`

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs [hunk 1]
```diff
@@ -112,8 +112,8 @@ public interface IModule : IEquatable<IModule>
         [JsonProperty(PropertyName = "imagePullPolicy")]
         ImagePullPolicy ImagePullPolicy { get; }
 
-        [JsonProperty(PropertyName = "priority")]
-        uint Priority { get; }
+        [JsonProperty(PropertyName = "startupOrder")]
+        uint StartupOrder { get; }
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs:116]: I thought we discussed that we only need to change the public facing name of this. The internal name can stay priority (to minimize the changes?)

Comment (yophilav) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs:116]: That's the original intent but it ripples through all the various classes and their tests.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs [hunk 2]
```diff
@@ -112,8 +112,8 @@ public interface IModule : IEquatable<IModule>
         [JsonProperty(PropertyName = "imagePullPolicy")]
         ImagePullPolicy ImagePullPolicy { get; }
 
-        [JsonProperty(PropertyName = "priority")]
-        uint Priority { get; }
+        [JsonProperty(PropertyName = "startupOrder")]
```
Comment (ancaantochi) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs:115]: it looks like the change is not backward compatible, what happens with customers using priority?

Comment (richma-ms) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs:115]: I don't think this went out for 1.0.9, so there shouldn't be anyone other than internal partners using this yet.

Comment (yophilav) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/IModule.cs:115]: https://github.com/Azure/azure-iotedge/releases/
Looks like the feature is first introduced in release 1.0.10-rc1

**Pull request merged**